src/pybind/mgr/dashboard_v2/README.rst

```diff
@@ -310,36 +416,62 @@ can be used:
 * ``pg_summary``: regular update of PG status information
 
 
-How to write a unit test when a controller accesses a Ceph module?
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+How to bound the response time of a controller that gets data from the cluster?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Consider the following example that implements a controller that retrieves the
-list of RBD images of the ``rbd`` pool::
+When your controller needs to access the Ceph cluster to retrieve some data,
+(for instance get the list of RADOS pools, or get the list of RBD images,
+etc.), you may want to limit the time that the frontend will wait for your
+controller's response, by returning the information that no data could be
+retrieved within a time interval, or by returning stale data retrieved in
+previous requests.
 
-  import rbd
-  from ..tools import ApiController, RESTController
+This way the frontend is in control of the requests timing and can warn or
+ask for feedback from the user, when such request takes longer than expected.
 
-  @ApiController('rbdimages')
-  class RbdImages(RESTController):
-      def __init__(self):
-          self.ioctx = self.mgr.rados.open_ioctx('rbd')
-          self.rbd = rbd.RBD()
+To help the developer in handling these kinds of situations we provide a
+function decorator called ``@ViewCache``, which the developer can use to
+decorate a function that makes accesses to the Ceph cluster and returns data.
 
-      def list(self):
-          return [{'name': n} for n in self.rbd.list(self.ioctx)]
+Example::
 
-In the example above, we want to mock the return value of the ``rbd.list``
-function, so that we can test the JSON response of the controller.
+  import rbd
+  from ..tools import ApiController, RESTController, ViewCache
 
-The unit test code will look like the following::
 
-  import mock
-  from .helper import ControllerTestCase
+  @ApiController('rbd')
+  class Rbd(RESTController):
+      def __init__(self):
+          super(Rbd, self).__init__()
+          self.rbd = rbd.RBD()
 
-  class RbdImagesTest(ControllerTestCase):
-      @mock.patch('rbd.RBD.list')
-      def test_list(self, rbd_list_mock):
-          rbd_list_mock.return_value = ['img1', 'img2']
-          self._get('/api/rbdimages')
-          self.assertJsonBody([{'name': 'img1'}, {'name': 'img2'}])
+      @ViewCache()
+      def image_list(self, pool_name):
```

Comment:
`image_list` can be `static`

Reply:
in this particular case no, because we are using a instance variable like `rbd`. Also, it's just an example.